### PR TITLE
added axios usage for filters.json

### PR DESCRIPTION
**Pull request by himanshu-spr**

### CODE REVIEW

File: src/components/FilterBar/index.tsx [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+import SearchFilter from "../SearchFilter";
+import { getFilters as getFiltersHelper } from "../../helpers/GetFilters";
+import "./FilterBar.css";
+import { DropDownData } from "../../interfaces";
 
 const FilterBar = () => {
-    return (
+  const [dropDownData, setDropDownData] = useState<DropDownData>(
+    {} as DropDownData
+  );
```
Comment (NiketMalik) [inline src/components/FilterBar/index.tsx]: `{} as DropDownData` why is this required?

File: src/helpers/GetFilters/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export const getFilters = async () => {
+  let data;
+  try {
+    const res = await axios.get("http://localhost:3000/filters.json");
+    data = res.data;
+  } catch (err) {
+    alert(err);
+  }
+
+  return data;
```
Comment (NiketMalik) [inline src/helpers/GetFilters/index.tsx]: Should throw here and catch where this is used

**Review by NiketMalik — CHANGES_REQUESTED** [review]

File: src/components/FilterBar/index.tsx (hunk 1 above)
Comment (himanshu-spr) [reply src/components/FilterBar/index.tsx]: {} will throw type error as it is not type of DropDownData

Comment (NiketMalik) [reply src/components/FilterBar/index.tsx:11]: your state type should be `DropDownData | null`

**Review by NiketMalik — CHANGES_REQUESTED** [review]

File: src/components/FilterBar/index.tsx (hunk 1 above)
Comment (NiketMalik) [reply src/components/FilterBar/index.tsx]: your state type should be `DropDownData | null`

File: src/components/FilterBar/index.tsx [hunk 3]
```diff
@@ -1,23 +1,43 @@
...
+import { DropDownData } from "../../interfaces";
 
 const FilterBar = () => {
-    return (
+  const [dropDownData, setDropDownData] = useState<DropDownData | null>(null);
+  const getData = async () => {
+    try {
+      const data = await getFiltersHelper();
+      setDropDownData((prevState) => {
+        return { ...prevState, ...data };
```
Comment (NiketMalik) [inline src/components/FilterBar/index.tsx]: You don't need to spread `prevState`, alwways have to override. Additionally this needs to be wrapped in `useCallback` and should be added as dep of `useEffect`

File: src/helpers/GetFilters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export const getFilters = async () => {
+  let data;
+  try {
+    const res = await axios.get("http://localhost:3000/filters.json");
+    data = res.data;
+  } catch (err) {
+    throw err;
+  }
+
+  return data;
```
Comment (NiketMalik) [inline src/helpers/GetFilters/index.tsx]:
```suggestion
  return axios.get("http://localhost:3000/filters.json").then(res => res.data)
```

**Review by NiketMalik — CHANGES_REQUESTED** [review]
